internal/fs/fs.go

```diff
@@ -1692,7 +1692,14 @@ func (fs *fileSystem) Rename(
 		}
 	}
 
-	// Find the object in the old location.
+	// If child is a local file inode (un-synced), rename operation is not supported.
+	localChild := fs.lookUpLocalFileInode(oldParent, op.OldName)
+	if localChild != nil {
+		defer localChild.Unlock()
```

Comment:
can you check if fs.unlockAndDecrementLookupCount(in, 1) as in renameDir is required here, since we are not sending response to the kernel.